### PR TITLE
Nav Unification: redirect to url when clicking a top level menu

### LINKED ISSUE #45963: Make top level menus redirect to their first children instead of just toggling the dropdown
In the scope of unifying the navigation between calypso and wp-admin we have to make the top level menus redirect to their first children instead of just toggling the dropdown, so that it works like it does in wp-admin.

eg. `Posts` should redirect and open `All Posts`

### PR DESCRIPTION
**Pull request by cpapazoglou**
#### Changes proposed in this Pull Request

* redirect user when clicking a top level menu if link exists

#### Testing instructions

* Run calyspo 
* Add ?flags=nav-unification to the url
* Click on `Posts` or `Pages` UI should load `posts` and `pages` accordingly
* Click on Setting, you should be redirected to wp-admin

SS 2020-09-29 at 18 33 16

Fixes #45963

### CODE REVIEW

**Comment by mreishus** [comment]
I noticed the PropType for `link` was missing, so I added it (along with one other missing proptype).

**Comment by mreishus** [comment]
Is this an issue?  I noticed that when I hover over *items*, my browser statusbar shows me where the link goes.  But when I hover over *items*, my statusbar doesn't show anything.

**Comment by obenland** [comment]
Something seems off there. Top level menu items should _not_ link to the first child's URL, but rather to the URL that is provided in API response. Just because Core-registered menu items have the same URL for the top-level and first child, doesn't mean that plugin-registered menu items will behave the same.

**Comment by cpapazoglou** [comment]
> Is this an issue? I noticed that when I hover over _items_, my browser statusbar shows me where the link goes. But when I hover over _menus_, my statusbar doesn't show anything.
> 
> 
> 
> 

@mreishus This is expected cause there is a JS rule for navigating to a menu a URL. Adding a proper `a` tag felt like over-engineering **at this point**.

> Top level menu items should not link to the first child's URL, but rather to the URL that is provided in API response. 

@obenland this PR **doesn't** attempt to redirect to the first child, it rather attempts to redirect to a link (provided by API) if it exists. Did you notice any wrong redirects?

**Comment by cpapazoglou** [comment]
Rebased with master as an attempt to make tests pass

**Review by getdave — APPROVED** [review]
I see this as a solid change.

Currently on `master` when you click a top level item all it does it expand the menu section. Nothing else. That is a strange experience (try it, it's disconcerting). Also that experience doesn't align well with WPAdmin where (testing on my personal website running latest WP ) when you click the `Posts` menu item it navigates and then displays with the submenus open.

I believe this PR gets it right namely clicking on `Posts` results in:

* Navigation to the Posts page view in Calypso.
* Expansion of the Posts menu to reveal all sub menu items.

From the code I see that there's nothing strange going on here. There's simply a conditional to check whether the menu item has a link and it is present it either:

* Navigates to a Calypso route and _then_ expands.
* Navigates to an external URL (and exits).

I also tested this and it worked exactly as described.

File: client/my-sites/sidebar-unified/menu.jsx [hunk 1]
```diff
@@ -20,15 +21,26 @@ import { toggleMySitesSidebarSection as toggleSection } from 'state/my-sites/sid
...
 	const isExpanded = useSelector( ( state ) => isSidebarSectionOpen( state, sectionId ) );
 
 	return (
 		<ExpandableSidebarMenu
-			onClick={ () => reduxDispatch( toggleSection( sectionId ) ) }
+			onClick={ () => {
+				if ( link ) {
+					if ( isExternal( link ) ) {
+						// If the URL is external, page() will fail to replace state between different domains
+						document.location.href = link;
```
Comment (tyxla) [inline client/my-sites/sidebar-unified/menu.jsx:37]: Nit: We have a `externalRedirect()` helper as part of `lib/route/path`, perhaps we could have used that?

Comment (cpapazoglou) [reply client/my-sites/sidebar-unified/menu.jsx:37]: Thanks for the heads up! I opened https://github.com/Automattic/wp-calypso/pull/46123

**Pull request merged**